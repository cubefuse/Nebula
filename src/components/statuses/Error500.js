//
import React from "react";
import { Layout } from "antd";

const { Content } = Layout;

/**
 * 404 page not found error page.
 */
export default class Error500 extends React.Component {
  render() {
    return (
      <Content style={{ textAlign: "center" }}>
        <h1 style={{ fontSize: "10em", color: "lightgray", marginBottom: 0 }}>
          500
        </h1>
        <h2 style={{ fontSize: "xx-large", color: "gray" }}>
          Ouch! This application has crashed.
        </h2>
        {this.props.error && <p>Debug: {this.props.error}</p>}
      </Content>
    );
  }
}
