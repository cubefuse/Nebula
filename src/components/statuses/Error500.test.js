import React from "react";
import "jest-dom/extend-expect";
import { render } from "react-testing-library";
import Error500 from "./Error500";

test("can render", () => {
  const node = render(<Error500 />);
  expect(node).toMatchSnapshot();
});

test("can render with error message", () => {
  const node = render(<Error500 error="This is a test debug message." />);
  expect(node).toMatchSnapshot();
});