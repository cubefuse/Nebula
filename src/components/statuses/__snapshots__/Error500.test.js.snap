// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`can render 1`] = `
Object {
  "container": <div>
    <div
      class="ant-layout-content"
      style="text-align: center;"
    >
      <h1
        style="font-size: 10em; margin-bottom: 0px;"
      >
        500
      </h1>
      <h2
        style="font-size: xx-large; color: gray;"
      >
        Ouch! This application has crashed.
      </h2>
    </div>
  </div>,
  "debug": [Function],
  "getAllByAltText": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getAllByValue": [Function],
  "getByAltText": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "getByValue": [Function],
  "queryAllByAltText": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryAllByValue": [Function],
  "queryByAltText": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "queryByValue": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`can render with error message 1`] = `
Object {
  "container": <div>
    <div
      class="ant-layout-content"
      style="text-align: center;"
    >
      <h1
        style="font-size: 10em; margin-bottom: 0px;"
      >
        500
      </h1>
      <h2
        style="font-size: xx-large; color: gray;"
      >
        Ouch! This application has crashed.
      </h2>
      <p>
        Debug: 
        This is a test debug message.
      </p>
    </div>
  </div>,
  "debug": [Function],
  "getAllByAltText": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getAllByValue": [Function],
  "getByAltText": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "getByValue": [Function],
  "queryAllByAltText": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryAllByValue": [Function],
  "queryByAltText": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "queryByValue": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
